FWCore/ParameterSet/test/test_catch_ParameterSetDescription.cc

```diff
@@ -1252,52 +1252,173 @@ TEST_CASE("test ParameterSetDescription", "[ParameterSetDescription]") {
+
+      templte.add<edm::InputTag>("t");
+      d.addParameter<edm::InputTag>("t", edm::InputTag("foo"));
+
+      templte.add<std::vector<edm::InputTag>>("vt");
+      d.addParameter<std::vector<edm::InputTag>>("vt", std::vector<edm::InputTag>({edm::InputTag("foo")}));
+
+      templte.add<edm::ESInputTag>("et");
+      d.addParameter<edm::ESInputTag>("et", edm::ESInputTag(":foo"));
+
+      templte.add<std::vector<edm::ESInputTag>>("vet");
+      d.addParameter<std::vector<edm::ESInputTag>>("vet", std::vector<edm::ESInputTag>({edm::ESInputTag(":foo")}));
+
+      templte.add<edm::EventID>("e");
+      d.addParameter<edm::EventID>("e", edm::EventID(1, 2, 3));
+
+      templte.add<std::vector<edm::EventID>>("ve");
+      d.addParameter<std::vector<edm::EventID>>("ve", std::vector<edm::EventID>({edm::EventID(1, 2, 3)}));
+
+      templte.add<edm::LuminosityBlockID>("L");
+      d.addParameter<edm::LuminosityBlockID>("L", edm::LuminosityBlockID(1, 2));
+
+      templte.add<std::vector<edm::LuminosityBlockID>>("vL");
+      d.addParameter<std::vector<edm::LuminosityBlockID>>(
+          "vL", std::vector<edm::LuminosityBlockID>({edm::LuminosityBlockID(1, 2)}));
+
+      templte.add<edm::EventRange>("er");
+      d.addParameter<edm::EventRange>("er", edm::EventRange(1, 2, 3, 4, 5, 6));
+
+      templte.add<std::vector<edm::EventRange>>("ver");
+      d.addParameter<std::vector<edm::EventRange>>("ver",
+                                                   std::vector<edm::EventRange>({edm::EventRange(1, 2, 3, 4, 5, 6)}));
+
+      templte.add<edm::LuminosityBlockRange>("Lr");
+      d.addParameter<edm::LuminosityBlockRange>("Lr", edm::LuminosityBlockRange(1, 2, 3, 4));
+
+      templte.add<std::vector<edm::LuminosityBlockRange>>("vLr");
+      d.addParameter<std::vector<edm::LuminosityBlockRange>>(
+          "vLr", std::vector<edm::LuminosityBlockRange>({edm::LuminosityBlockRange(1, 2, 3, 4)}));
+
+      psetDesc.addVPSet("vp", templte, defaults);
     }
+    edm::ParameterSet test;
+    test.addParameter("vp", defaults);
+    psetDesc.validate(test);
+
+    std::ostringstream s;
+    psetDesc.writeCfi(s, false, 0);
+    std::string expected = R"-(
+vp = cms.VPSet(
+  cms.PSet(
+    L = cms.LuminosityBlockID(1, 2),
+    Lr = cms.LuminosityBlockRange('1:2-3:4'),
+    b = cms.bool(True),
+    d = cms.double(1),
+    e = cms.EventID(1, 2, 3),
+    er = cms.EventRange('1:2:3-4:5:6'),
+    et = cms.ESInputTag('', 'foo'),
+    i = cms.int32(1),
+    l = cms.int64(1),
+    t = cms.InputTag('foo'),
+    ui = cms.uint32(1),
+    ul = cms.uint64(1),
+    vL = cms.VLuminosityBlockID('1:2'),
+    vLr = cms.VLuminosityBlockRange('1:2-3:4'),
+    vd = cms.vdouble(1),
+    ve = cms.VEventID('1:2:3'),
+    ver = cms.VEventRange('1:2:3-4:5:6'),
+    vet = cms.VESInputTag(':foo'),
+    vi = cms.vint32(1),
+    vl = cms.vint64(1),
+    vt = cms.VInputTag('foo'),
+    vui = cms.vuint32(1),
+    vul = cms.vuint64(1)
```

Comment:
I see `string`, `vstring`, `FileInPath`, `vbool` are "missing" from this list. Are they tested well-enough elsewhere, or would there be value in adding them here?

(in principle also inner `PSet` and `VPSet` are not there, but maybe they are tested already well-enough elsewhere?)

Reply:
We don't have a `vbool` in the system.  Seems FileInPath's can't be created without the file being there (there is a special test in this unit test working around that).  I'll definitely add the `string` and `vstring`.

Reply:
I was able to add all the presently available options to the test.